aten/src/ATen/cpu/vec256/vsx/vsx_helpers.h

```diff
@@ -0,0 +1,326 @@
+      const {                                                            \
+    using vvtype = Vec256<op_type>::vec_internal_type;                   \
+    const vvtype v_one = vec_splats(static_cast<op_type>(1.0));          \
+    vvtype ret0 = (vvtype)func(_vec0, other._vec0);                      \
+    vvtype ret1 = (vvtype)func(_vec1, other._vec1);                      \
+    return Vec256<op_type>{vec_and(ret0, v_one), vec_and(ret1, v_one)};  \
+  }
+
+#define DEFINE_CLAMP_FUNCS(operand_type)                                \
+  template <>                                                           \
+  Vec256<operand_type> inline __inline_attrs clamp(                     \
+      const Vec256<operand_type>& a, const Vec256<operand_type>& min,   \
+      const Vec256<operand_type>& max) {                                \
+    return Vec256<operand_type>{                                        \
+        vec_min(max.vec0(), vec_max(a.vec0(), min.vec0())),             \
+        vec_min(max.vec1(), vec_max(a.vec1(), min.vec1()))};            \
+  }                                                                     \
+  template <>                                                           \
+  Vec256<operand_type> inline __inline_attrs clamp_min(                 \
+      const Vec256<operand_type>& a, const Vec256<operand_type>& min) { \
+    return Vec256<operand_type>{vec_max(a.vec0(), min.vec0()),          \
+                                vec_max(a.vec1(), min.vec1())};         \
+  }                                                                     \
+  template <>                                                           \
+  Vec256<operand_type> inline __inline_attrs clamp_max(                 \
+      const Vec256<operand_type>& a, const Vec256<operand_type>& max) { \
+    return Vec256<operand_type>{vec_min(a.vec0(), max.vec0()),          \
+                                vec_min(a.vec1(), max.vec1())};         \
+  }
+
+#define DEFINE_REINTERPRET_CAST_FUNCS(first_type, cast_type,           \
+                                      cast_inner_vector_type)          \
+  template <>                                                          \
+  inline __inline_attrs Vec256<cast_type> cast<cast_type, first_type>( \
+      const Vec256<first_type>& src) {                                 \
+    return Vec256<cast_type>{(cast_inner_vector_type)src.vec0(),       \
+                             (cast_inner_vector_type)src.vec1()};      \
+  }
+
+#define DEFINE_REINTERPRET_CAST_TO_ALL_FUNCS(first_type)     \
+  DEFINE_REINTERPRET_CAST_FUNCS(first_type, double, __vd)    \
+  DEFINE_REINTERPRET_CAST_FUNCS(first_type, float, __vf)     \
+  DEFINE_REINTERPRET_CAST_FUNCS(first_type, int64_t, __vlli) \
+  DEFINE_REINTERPRET_CAST_FUNCS(first_type, int32_t, __vi)   \
+  DEFINE_REINTERPRET_CAST_FUNCS(first_type, int16_t, __vshi)
+
+// it can be used to emulate blend faster
+constexpr int blendChoice(int mask, int half1 = 0xF, int half2 = 0xF0) {
```

Comment:
Are you sure you don't wanna use fixed-width integers here?

Reply:
which line?

Reply:
if its about constexpr then I dont see any reason. But I will recheck again

Reply:
I meant `mask` throughout this file

Reply:
I will check masks  range again. As guaranted int is 2bytes at least. Probably it be safer to make it int32_t